src/lib/url.js

```diff
@@ -27,7 +67,7 @@ export const parseJsonApiUrl = fromUrl => {
     port: url.port,
     path: url.pathname,
     query: {
-      filter: query.get('filter'),
+      filter: parseQueryParameterFamily('filter', query.entries()),
```

Comment:
This is exactly what I was suggesting in the last PR, rewriting this function to use the more robust query destructuring. Really nice work!